pilot/pkg/security/model/authentication.go

```diff
@@ -128,17 +128,34 @@ func AppendURIPrefixToTrustDomain(trustDomainAliases []string) []string {
 // ApplyToCommonTLSContext completes the commonTlsContext
 func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Proxy,
 	subjectAltNames []string, crl string, trustDomainAliases []string, validateClient bool,
+	tlsCertificates []*networking.ServerTLSSettings_TLSCertificate,
 ) {
+	sdsSecretConfigs := make([]*tls.SdsSecretConfig, 0)
 	customFileSDSServer := proxy.Metadata.Raw[security.CredentialFileMetaDataName] == "true"
+	// Envoy does not support client validation using multiple CA certificates when multiple certificates are provided.
+	// So we only use what's provided in the ServerTLSSettings.
+	caCert := proxy.Metadata.TLSServerRootCert
```

Comment:
what's the relationship with ServerTLSSettings_TLSCertificate.CaCertificate?

Reply:
The CA cert specified in ServerTLSSettings_TLSCertificate is used to build ValidationContext for both the the certs specified in TLSCertificates array.

Reply:
The CA cert specified in ServerTLSSettings_TLSCertificate is used to build ValidationContext for both the the certs specified in TLSCertificates array.

Reply:
And with serverTLSSettings.TlsCertificates, which takes higher priority? 

>Envoy does not support client validation using multiple CA certificates when multiple certificates are provided.

Where is it from?

Reply:
In the [common tls context](https://www.envoyproxy.io/docs/envoy/v1.34.0/api-v3/extensions/transport_sockets/tls/v3/tls.proto#envoy-v3-api-msg-extensions-transport-sockets-tls-v3-commontlscontext) sds_secret_config is an array  so you can supply multiple tls certificates and in the validation context it is single  - so you can only supply single ca cert. I checked and it seems generally that is the case CAs operate - they will have one signing cert for multiple types of certs issued

Reply:
Seems this is a limitation from envoy, actually maybe you can merge the two ca root cert into one file. And then you can use it to verify clients

Reply:
Don't know. That is a good suggestion. We can try